client/src/components/pub-entry.jsx

```diff
@@ -0,0 +1,104 @@
+import React, { Component } from 'react';
+// import { Redirect } from '@reach/router';
+import { ApiService } from '../services';
+
+class PubEntry extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: '',
+      twitterHandle: '',
+      entryTitle: '',
+      entryTwitterHandle: '@',
+    };
+  }
+
+  handleInputChange = (event) => {
+    const { name, value } = event.target;
+    this.setState({
+      [name]: value,
+    });
+  };
+
+  handleChange = (event) => {
+    let target = event.target;
+    let name = target.name;
+    this.setState({
+      [name]: event.target.value,
+    });
+  };
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+    if (this.state.entryTitle.length === 0 || this.state.entryTwitterHandle.length === 1) {
+      alert('Please enter a valid name and Twitter handle for the publication');
+      window.location.replace('/pub-entry');
+    } else {
+      ApiService.post('pubs', {
+        name: this.state.entryTitle,
+        twitterHandle: this.state.entryTwitterHandle,
+      })
+        .then((response) => {
+          if (response.statusText === 'OK') {
```

Comment:
I built this response validation because I was having problems in my Timekeeper app with the page reloading to the destination (1) sometimes before the data was written to the db and (2) almost always before the updated data was pulled from the db. This fixed it. Maybe `navigate` solves that problem? Or maybe there's a better way to do this. What would you suggest here?

Reply:
I think this validation logic is a “code smell” of a different issue. It’s fine to leave it but it otherwise should not be necessary.

Reply:
Now I know what 'code smell' means... I will take it out and we'll see what the effect is. Maybe we can discuss the problem at some point? (that list is getting long...)